src/include/internal/generic.hpp

```diff
@@ -18,7 +18,9 @@
 
 // Boost
 #include <boost/math/constants/constants.hpp>
-#include <gsl/gsl_blas.h>
+// Eigen
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/Dense>
```

Comment:
Ideally these shouldn't be included in multiple places.

Reply:
Yup, maybe in a followup.